tests/test_specials.py::test_slash_l* fail with non-en_US locale

We're running the test suite with `C.UTF-8` locale (because unlike `en_US` it's guaranteed to be available on all Gentoo systems), and this results in the following test failures:

```pytb
============================================================== FAILURES ===============================================================
____________________________________________________________ test_slash_l _____________________________________________________________

executor = <function executor.<locals>.query_runner at 0x7f7cfb8154e0>

    @dbtest
    def test_slash_l(executor):
        results = executor(r"\l")
        row = ("_test_db", "postgres", "UTF8", "en_US.UTF-8", "en_US.UTF-8", None)
        headers = ["Name", "Owner", "Encoding", "Collate", "Ctype", "Access privileges"]
>       assert row in results[1]
E       AssertionError: assert ('_test_db', 'postgres', 'UTF8', 'en_US.UTF-8', 'en_US.UTF-8', None) in [('_test_db', 'postgres', 'UTF8', 'C', 'C.UTF8', None), ('postgres', 'postgres', 'UTF8', 'C', 'C.UTF8', None), ('template0', 'postgres', 'UTF8', 'C', 'C.UTF8', '=c/postgres\npostgres=CTc/postgres'), ('template1', 'postgres', 'UTF8', 'C', 'C.UTF8', '=c/postgres\npostgres=CTc/postgres')]

executor   = <function executor.<locals>.query_runner at 0x7f7cfb8154e0>
headers    = ['Name', 'Owner', 'Encoding', 'Collate', 'Ctype', 'Access privileges']
results    = [None,
 [('_test_db', 'postgres', 'UTF8', 'C', 'C.UTF8', None),
  ('postgres', 'postgres', 'UTF8', 'C', 'C.UTF8', None),
  ('template0',
   'postgres',
   'UTF8',
   'C',
   'C.UTF8',
   '=c/postgres\npostgres=CTc/postgres'),
  ('template1',
   'postgres',
   'UTF8',
   'C',
   'C.UTF8',
   '=c/postgres\npostgres=CTc/postgres')],
 ['Name', 'Owner', 'Encoding', 'Collate', 'Ctype', 'Access privileges'],
 'SELECT 4']
row        = ('_test_db', 'postgres', 'UTF8', 'en_US.UTF-8', 'en_US.UTF-8', None)

tests/test_specials.py:16: AssertionError
________________________________________________________ test_slash_l_pattern _________________________________________________________

executor = <function executor.<locals>.query_runner at 0x7f7cfb8163e0>

    @dbtest
    def test_slash_l_pattern(executor):
        results = executor(r"\l _test*")
        row = [("_test_db", "postgres", "UTF8", "en_US.UTF-8", "en_US.UTF-8", None)]
        headers = ["Name", "Owner", "Encoding", "Collate", "Ctype", "Access privileges"]
>       assert row == results[1]
E       AssertionError: assert [('_test_db', 'postgres', 'UTF8', 'en_US.UTF-8', 'en_US.UTF-8', None)] == [('_test_db', 'postgres', 'UTF8', 'C', 'C.UTF8', None)]
E         At index 0 diff: ('_test_db', 'postgres', 'UTF8', 'en_US.UTF-8', 'en_US.UTF-8', None) != ('_test_db', 'postgres', 'UTF8', 'C', 'C.UTF8', None)
E         Full diff:
E         - [('_test_db', 'postgres', 'UTF8', 'C', 'C.UTF8', None)]
E         ?                                    ^^^^^^
E         + [('_test_db', 'postgres', 'UTF8', 'en_US.UTF-8', 'en_US.UTF-8', None)]
E         ?                                    ^^^^^    +   +++++++++++++++

executor   = <function executor.<locals>.query_runner at 0x7f7cfb8163e0>
headers    = ['Name', 'Owner', 'Encoding', 'Collate', 'Ctype', 'Access privileges']
results    = [None,
 [('_test_db', 'postgres', 'UTF8', 'C', 'C.UTF8', None)],
 ['Name', 'Owner', 'Encoding', 'Collate', 'Ctype', 'Access privileges'],
 'SELECT 1']
row        = [('_test_db', 'postgres', 'UTF8', 'en_US.UTF-8', 'en_US.UTF-8', None)]

tests/test_specials.py:25: AssertionError
```

Could you perhaps use the values from `LANG`/`LC_*` variables rather than hardcoding `en_US`? Also, the canonical IANA encoding name is `UTF-8`, not `UTF8` — using the latter could result in weird problems e.g. on BSD systems.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -54,6 +54,8 @@ jobs:
           pip install -r requirements-dev.txt
 
       - name: Run unit tests
+        env:
+          LANG: en_US.UTF-8
         run: coverage run --source pgspecial -m pytest
 
       - name: Run Black

--- a/changelog.rst
+++ b/changelog.rst
@@ -2,6 +2,7 @@ Upcoming
 ========
 
 * Added `build-system` section to `pyproject.toml` to use the modern setuptools backend.
+* Fixed test failures when locale other than `en_US` is used.
 
 2.1.0 (2023-03-31)
 =========

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,6 +1,10 @@
 import pytest
 from dbutils import create_db, db_connection, setup_db, teardown_db, TEST_DB_NAME
+import locale
 from pgspecial.main import PGSpecial
+
+
+locale.setlocale(locale.LC_ALL, "")
 
 
 @pytest.fixture(scope="module")

--- a/tests/test_specials.py
+++ b/tests/test_specials.py
@@ -4,14 +4,24 @@
 import pytest
 from dbutils import dbtest, POSTGRES_USER, SERVER_VERSION, foreign_db_environ, fdw_test
 import itertools
+import locale
 
 objects_listing_headers = ["Schema", "Name", "Type", "Owner", "Size", "Description"]
+
+# note: technically, this is the database encoding, not the client
+# locale but for the purpose of testing we can assume the server
+# and the client are using the same locale
+# note 2: locale.getlocale() does not return the raw values,
+# in particular it transforms C into en_US, so we use .setlocale()
+# instead as that matches the C library function
+LC_COLLATE = locale.setlocale(locale.LC_COLLATE, None)
+LC_CTYPE = locale.setlocale(locale.LC_CTYPE, None)
 
 
 @dbtest
 def test_slash_l(executor):
     results = executor(r"\l")
-    row = ("_test_db", "postgres", "UTF8", "en_US.UTF-8", "en_US.UTF-8", None)
+    row = ("_test_db", "postgres", "UTF8", LC_COLLATE, LC_CTYPE, None)
     headers = ["Name", "Owner", "Encoding", "Collate", "Ctype", "Access privileges"]
     assert row in results[1]
     assert headers == results[2]
@@ -20,7 +30,7 @@ def test_slash_l(executor):
 @dbtest
 def test_slash_l_pattern(executor):
     results = executor(r"\l _test*")
-    row = [("_test_db", "postgres", "UTF8", "en_US.UTF-8", "en_US.UTF-8", None)]
+    row = [("_test_db", "postgres", "UTF8", LC_COLLATE, LC_CTYPE, None)]
     headers = ["Name", "Owner", "Encoding", "Collate", "Ctype", "Access privileges"]
     assert row == results[1]
     assert headers == results[2]

